server/src/sas/CodeZoneManager.ts

```diff
@@ -394,7 +394,7 @@ export class CodeZoneManager {
     } while (/^\s*$/.test(text));
 
     if (Lexer.isComment[type] || Lexer.isLiteral[type]) {
-      token = this._token(context.line, col + 1)!;
+      token = this._token(context.line, col - 1)!;
```

Comment:
@jingjiajie, would you please provide a detailed sample about why need to change this?
As it is fundamental logic, I want to verify it carefully. Thank you.

Reply:
Here is an example. The "e" of "endinteractive;" is misidentified if not change the logic of line 259, 397. Due to code zone is misidentified.

```
proc python;
  interactive;
  x = '''
    endinteractive
  '''
  print('hello') 
endinteractive;
run;
```

![a](https://github.com/user-attachments/assets/df05f89e-8ebd-4f0e-a386-dc6b95a6e323)